src/lib/Microsoft.Health.Events/EventConsumers/Service/EventBatchingService.cs

```diff
@@ -23,6 +23,10 @@ public class EventBatchingService : IEventConsumerService
     {
         private ConcurrentDictionary<string, EventPartition> _eventPartitions;
         private int _maxEvents;
+        private int? _scaledOutMaxEventsAllPartitions;
+        private int _scaledOutMaxEventsPerPartition;
+        private static object _lock = new object();
```

Comment:
Suggestion:  While you can lock I believe using the lighter weight interlock is more appropriate. See https://learn.microsoft.com/en-us/dotnet/api/system.threading.interlocked?view=net-7.0.  It looks like you just need to read, increment, or exchange values.  If so, the interlock corresponding options will work.  If you have multiple statements in the lock that need to be preserved as an atomic section then interlock wouldn't be a good solution.  It didn't appear you had any scenario like that but I could be mistaken.

Reply:
In reviewing the code it seems you only have one section where both are used, ChangeMaximumEventsPerPartition.  You could do Interlock.Reads on both then interlock exchange on the max events but that would run the risk of potentially an invalid buffer if we have had two partition changes back to back and the last one finished processing first.  So in this case the lock would be appropriate.  Thanks for the explanation.